cpmpy/expressions/python_builtins.py

```diff
@@ -100,3 +102,61 @@ def sum(iterable):
     if not any(isinstance(elem, Expression) for elem in iterable):
         return np.sum(iterable)
     return Operator("sum", iterable)
+
+@hook(bool, int)
+def __and__(self, other):
+    if not (isinstance(other,_IntVarImpl) or isinstance(other, _BoolVarImpl) ):
+        return orig(self, other)
+    else:
+
+        if not self.is_bool():
+            raise CPMpyException(f"Logical conjunction involving an integer ({self}) is not allowed")
+
+        if isinstance(self,int):
```

Comment:
Lets wait for this until we cleared out how to deal with ints and logical operators...

Reply:
Yes true